test/kubemark/gce/util.sh

```diff
@@ -80,6 +81,30 @@ function create-kubemark-master {
             "${KUBE_ROOT}/hack/e2e-internal/e2e-grow-cluster.sh"
         done
     fi
+
+    # If cluster uses private master IP, two kubeconfigs are created:
+    # - kubeconfig with public IP, which will be used to connect to the cluster
+    #     from outside of the cluster network
+    # - kubeconfig with private IP (called internal kubeconfig), which will be
+    #      used to create hollow nodes.
+    #
+    # Note that hollow nodes might use either of these kubeconfigs, but
+    # using internal one is better from performance and cost perspective, since
+    # traffic does not need to go through Cloud NAT.
+    REGION=$(grep -o "^[a-z]*-[a-z0-9]*" "$KUBE_TEMP"/cluster-location.txt)
```

Comment:
Isn't REGION already set:
https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/config-test.sh#L27
?

Reply:
Discussed offline, but for reference:

If zonal cluster is created, at this point we only have ZONE available. This is because the e2e-up.sh script is not sourced, just executed so we don't have access to vars set by it.

Also added a comment describing it more clearly.